src/fetch/object_id.rs

```diff
@@ -60,6 +67,69 @@ where
     Kind: Object,
     for<'de2> <Kind as Object>::Kind: serde::Deserialize<'de2>;
 
+/// get a cache map for a specific object kind
+/// creates a new cache map if not already exists
+async fn get_cache_for_type<Kind>() -> Cache<Url, Result<Kind, Kind::Error>>
+where
+    Kind: Object + Send + Sync + Clone + 'static,
+    <Kind as Object>::Error: Clone + Send + Sync,
+{
+    static CACHES: OnceLock<RwLock<type_map::concurrent::TypeMap>> = OnceLock::new();
```

Comment:
Does this really need two different types of lock?

Reply:
The reason is that

1. OnceLock::get_or_init() returns an immutable reference to the inner value. It only cares about making sure the initializer is called only once.
2. TypeMap is basically just a HashMap, which cannot be written to concurrently. (the concurrent in the package name just refers to the inner bounds of the value type of the typemap.
3. RwLock allows writing to the TypeMap via reference.

Honestly I just spent a while trying to figure out the best way to do this in general, I'd like it to not need a typemap at all (somehow abuse monomorphization to create one moka cache per type) but couldn't figure out a better solution.